pooch/downloaders.py

```diff
@@ -49,6 +53,10 @@ def choose_downloader(url):
         "https": HTTPDownloader,
         "http": HTTPDownloader,
     }
+
+    if paramiko is not None:
+        known_downloaders.update({"sftp": SFTPDownloader})
```

Comment:
Having this will make the error message  strange because we list `SFTPDownloader` as a part of the API and it's not obvious why a user would be "Unrecognized URL protocol sftp..." if that's the case. So it can be included permanently in `known_downloaders` and we can let the class itself raise the exception.